### PR TITLE
Feature/add initial AR-PARAMETER support for SW-AXIS-GROUPED

### PR DESCRIPTION
**Pull request by acomodi**

### CODE REVIEW

**Comment by acomodi** [comment]
[arxml support for AR-PARAMETER tag](https://trello.com/c/wFZWoOcv/789-arxml-support-for-ar-parameter-tag)

File: autosar/parser/parser_base.py [hunk 1]
```diff
@@ -402,14 +402,29 @@ def parseSwAxisGrouped(self, rootXML, parent = None):
                 sharedAxisTypeRef = self.parseTextNode(itemXML)
             elif tag == "SW-AXIS-INDEX":
                 swAxisIndex = self.parseTextNode(itemXML)
+            elif tag == "AR-PARAMETER":
```
Comment (andredami) [inline autosar/parser/parser_base.py:405]: Here you are managing `<AR-PARAMETER>` in `<SW-AXIS-GROUPED>`.
From the xml schema, `<AR-PARAMETER>`s (defined in `SW-CALPRM-REF-PROXY` complex type) are also possibile in `<SW-DATA-DEPENDENCY-ARGS>` for which I can't find support.
`<SW-DATA-DEPENDENCY-ARGS>` are part of `<SW-DATA-DEPENDENCY>`, which is part of `<SW-DATA-DEF-PROPS-CONTENT>`, which is defined as part of the complex type `SW-DATA-DEF-PROPS-CONDITIONAL`, used in **`<SW-DATA-DEF-PROPS>`** (element from which `<SW-AXIS-GROUPED>` depends as well).

I think we should support both or at least leave a TODO in the code and submit a task to track the partial support.

File: autosar/parser/parser_base.py [hunk 2]
```diff
@@ -402,14 +402,29 @@ def parseSwAxisGrouped(self, rootXML, parent = None):
...
+            swAxisIndex=swAxisIndex,
+            accessedParameter=accessedParameter
         )
 
+    def parseParameterInstanceRef(self, xmlRoot):
+        """parses <AUTOSAR-PARAMETER-IREF>"""
+        assert(xmlRoot.tag == 'AUTOSAR-PARAMETER-IREF')
+        portRef = self.parseTextNode(xmlRoot.find('PORT-PROTOTYPE-REF'))
+        parameterDataRef = self.parseTextNode(xmlRoot.find('TARGET-DATA-PROTOTYPE-REF'))
+        return autosar.behavior.ParameterInstanceRef(portRef, parameterDataRef)
```
Comment (andredami) [inline autosar/parser/parser_base.py:427]: Missing `<ROOT-PARAMETER-DATA-PROTOTYPE-REF>` and `<CONTEXT-DATA-PROTOTYPE-REF>`

**Review by andredami — CHANGES_REQUESTED** [review]

**Review by andredami — APPROVED** [review]
LGTM

**Pull request merged**